src/config/prettier-config-explorer/common.js

```diff
@@ -0,0 +1,46 @@
+import fs from "node:fs/promises";
+
+const CONFIG_FILE_NAMES = [
+  "package.json",
+  ".prettierrc",
+  ".prettierrc.json",
+  ".prettierrc.yaml",
+  ".prettierrc.yml",
+  ".prettierrc.json5",
+  ".prettierrc.js",
+  ".prettierrc.mjs",
+  ".prettierrc.cjs",
+  "prettier.config.js",
+  "prettier.config.mjs",
+  "prettier.config.cjs",
+  ".prettierrc.toml",
+];
+
+async function fileExists(file) {
+  let stats;
+  try {
+    stats = await fs.stat(file);
+  } catch {
+    return false;
+  }
+
+  return stats.isFile();
+}
+
+function createCachedFunction(function_) {
+  const cache = new Map();
+
+  return async function (fileOrDirectory) {
+    if (cache.has(fileOrDirectory)) {
+      return cache.get(fileOrDirectory);
+    }
+
+    const promise = function_(fileOrDirectory);
+    cache.set(fileOrDirectory, promise);
+    const result = await promise;
+    cache.set(fileOrDirectory, result);
+    return result;
+  };
+}
```

Comment:
This function is only used in `config-searcher.js`, right? I prefer to move from `common.js` to `config-searcher.js` and name the function something like `createCachedSearchFunction`. The name createCachedFunction is too generic.